asset-transfer-sbe/application-javascript/app.js

```diff
@@ -0,0 +1,354 @@
+async function main() {
+	try {
+		// use a random key so that we can run multiple times
+		const assetKey = `asset-${Math.floor(Math.random() * 100) + 1}`;
+
+		/** ******* Fabric client init: Using Org1 identity to Org1 Peer ******* */
+		const gatewayOrg1 = await initGatewayForOrg1();
+		const networkOrg1 = await gatewayOrg1.getNetwork(channelName);
+		const contractOrg1 = networkOrg1.getContract(chaincodeName);
+
+		/** ******* Fabric client init: Using Org2 identity to Org2 Peer ******* */
+		const gatewayOrg2 = await initGatewayForOrg2();
+		const networkOrg2 = await gatewayOrg2.getNetwork(channelName);
+		const contractOrg2 = networkOrg2.getContract(chaincodeName);
+
+		try {
+			let result;
+			let transaction;
+
+			try {
+				// Create an asset by organization Org1, this will require that both organization endorse.
+				// The endorsement will be handled by Discovery, since the gateway was connected with discovery enabled.
+				console.log(`\n--> Submit Transaction: CreateAsset, ${assetKey} as Org1 - endorsed by Org1 and Org2`);
+				await contractOrg1.submitTransaction('CreateAsset', assetKey, '100', 'Tom');
+				console.log('*** Result: committed, now asset will only require Org1 to endorse');
+			} catch (createError) {
+				console.log(`*** Failed: create - ${createError}`);
+			}
+
+			await readAsset(assetKey, 100, org1, contractOrg1, contractOrg2);
+
+			try {
+				// Since the gateway is using discovery we should limit the organizations used by
+				// discovery to endorse. This way we only have to know the organization and not
+				// the actual peers that may be active at any given time.
+				console.log(`\n--> Submit Transaction: UpdateAsset ${assetKey}, as Org1 - endorse by Org1`);
+				transaction = contractOrg1.createTransaction('UpdateAsset');
+				transaction.setEndorsingOrganizations(org1);
+				result = await transaction.submit(assetKey, '200');
+				console.log('*** Result: committed');
+			} catch (updateError) {
+				console.log(`*** Failed: update - ${updateError}`);
+			}
+
+			await readAsset(assetKey, 200, org1, contractOrg1, contractOrg2);
+
+			try {
+				// Submit a transaction to make an update to the asset that has a key-level endorsement policy
+				// set to only all Org1 to make updates. The following example will not use the "setEndorsingOrganizations"
+				// to limit the organizations that will do the endorsement, this means that it will be sent to all
+				// organizations in chaincode endorsement policy. Org1 will endorse, Org2 will not endorse, however
+				// since the Org1 approves and it is the only one required, the transaction will be committed.
+				console.log(`\n--> Submit Transaction: UpdateAsset ${assetKey}, as Org1 - endorse by Org1 and Org2`);
+				transaction = contractOrg1.createTransaction('UpdateAsset');
+				result = await transaction.submit(assetKey, '300');
+				console.log('*** Result: committed - because Org1 was on the discovery list, Org2 did not endorse');
```

Comment:
```suggestion
				console.log('*** Result: committed - because Org1 and Org2 both endorsed, while only the Org1 endorsement was required and checked');
```

Reply:
done